### DISCUSSION TITLE
How to use useContext hook in react?

### DISCUSSION THREAD

**Discussion by starlitnightsky** [starlitnightsky/Real-Estate#2] [Q&A] [answered]
Hello, everyone.
How to use useContext hook in react?

**Comment by wannacfuture** [comment, marked as answer]
The useContext hook in React allows you to consume values from a React context. Here are the basic steps to use it:

First, you need to create a context using the createContext method.

import React from 'react';

const MyContext = React.createContext();
Then, you can define your provider component that will set the value of the context.
jsx
function MyProvider(props) {
  const [value, setValue] = React.useState('default value');
  
  return (
    <MyContext.Provider value={[value, setValue]}>
      {props.children}
    </MyContext.Provider>
  )
}
In this case, the provider sets the initial value of the context to 'default value'.

Finally, you can use the useContext hook in any child component that needs to access the value of the context.

function MyChildComponent() {
  const [value, setValue] = React.useContext(MyContext);
  
  return (
    <div>
      <p>The current value is: {value}</p>
      <button onClick={() => setValue('new value')}>
        Change value
      </button>
    </div>
  );
}

**Reply by starlitnightsky** [reply]
Good. Thank you.
